dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/export/trackerevent/JdbcTrackerEventStore.java

```diff
@@ -987,151 +979,23 @@ private StringBuilder getFromWhereClause(
     return fromBuilder;
   }
 
-  private String getOrgUnitSql(
-      TrackerEventQueryParams params, User user, MapSqlParameterSource mapSqlParameterSource) {
-    return switch (params.getOrgUnitMode()) {
-      case CAPTURE -> createCaptureSql(user, mapSqlParameterSource);
-      case ACCESSIBLE -> createAccessibleSql(user, params, mapSqlParameterSource);
-      case DESCENDANTS -> createDescendantsSql(user, params, mapSqlParameterSource);
-      case CHILDREN -> createChildrenSql(user, params, mapSqlParameterSource);
-      case SELECTED -> createSelectedSql(user, params, mapSqlParameterSource);
-      case ALL -> null;
-    };
-  }
-
-  private String createCaptureSql(User user, MapSqlParameterSource mapSqlParameterSource) {
-    return createCaptureScopeQuery(user, mapSqlParameterSource, "");
-  }
-
-  private String createAccessibleSql(
-      User user, TrackerEventQueryParams params, MapSqlParameterSource mapSqlParameterSource) {
-
-    if (isProgramRestricted(params.getProgram()) || isUserSearchScopeNotSet(user)) {
-      return createCaptureSql(user, mapSqlParameterSource);
-    }
-
-    mapSqlParameterSource.addValue(COLUMN_USER_UID, user.getUid());
-    return getSearchAndCaptureScopeOrgUnitPathMatchQuery(
-        user, USER_SCOPE_ORG_UNIT_PATH_LIKE_MATCH_QUERY);
-  }
-
-  private String createDescendantsSql(
-      User user, TrackerEventQueryParams params, MapSqlParameterSource mapSqlParameterSource) {
-    mapSqlParameterSource.addValue(COLUMN_ORG_UNIT_PATH, params.getOrgUnit().getStoredPath());
+  private String addOrgUnitSql(
+      TrackerEventQueryParams params, MapSqlParameterSource sqlParameters, SqlHelper hlp) {
+    StringBuilder orgUnitBuilder = new StringBuilder();
 
-    if (isProgramRestricted(params.getProgram())) {
-      return createCaptureScopeQuery(
-          user, mapSqlParameterSource, AND + CUSTOM_ORG_UNIT_PATH_LIKE_MATCH_QUERY);
+    StringBuilder modeBuilder = new StringBuilder();
+    if (params.getOrgUnit() != null) {
+      buildOrgUnitModeClause(
```

Comment:
Do you already know `!Strings.isNullOrEmpty(modeBuilder.toString())` is true in this case? If so you can `orgUnitBuilder.append(hlp.whereAnd())` in here and pass `orgUnitBuilder` to `buildOrgUnitModeClause` 🤔